library/std/src/io/mod.rs

```diff
@@ -945,6 +945,33 @@ pub trait Read {
     }
 }
 
+/// Convenience function for [`Read::read_to_string`].
+///
+/// This avoids having to create a variable first and it provides more type safety
+/// since you can only get the buffer out if there were no errors. (If you use
+/// [`Read::read_to_string`] you have to remember to check whether the read succeeded
+/// because otherwise your buffer will be empty.)
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(io_read_to_string)]
+///
+/// # use std::io;
+/// fn main() -> io::Result<()> {
+///     let stdin = io::read_to_string(&mut io::stdin())?;
+///     println!("Stdin was:");
+///     println!("{}", stdin);
+///     Ok(())
+/// }
+/// ```
+#[unstable(feature = "io_read_to_string", issue = "80218")]
+pub fn read_to_string<R: Read>(reader: &mut R) -> Result<String> {
+    let mut buf = String::new();
```

Comment:
How large a buffer to pre-allocate before reading ?

In [std::fs::read_to_string](https://doc.rust-lang.org/src/std/fs.rs.html#275-283), we can allow a suitable buffer based on the size of the file to reduce the number of memory allocations. Here, we can't estimate how much is going to read the data.

Reply:
I think any type implementing `Read` itself can call `buf.reserve(len)`, there's no need to do it here - especially since, like you said, there's no way to know the right capacity.